### PR TITLE
Override NoDiscovery to true in node config

### PR DESCRIPTION
**Pull request by pedropombeiro**

### CODE REVIEW

**Comment by ghost** [comment]
### Pull Request Checklist
- [x] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?

**Review by jakubgs — APPROVED** [review]
Thanks!

**Comment by jakubgs** [comment]
Wait, I'm testing this change and it still fails:
```bash
 % ./build/bin/node-canary -mailserver enode://015e22f6cd2b44c8a51bd7a23555e271e0759c7d7f52432719665a74966f2da456d28e154e836bee6092b4d686fe67e331655586c57b718be3997c1629d24167@35.226.21.19:30504
ERROR[10-27|15:29:15.608] Node start failed                        package=status-go/cmd/node-canary error="NoDiscovery is false, but ClusterConfig.BootNodes is empty"
```

File: config/cli/fleet-eth.test.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "LogLevel": "DEBUG",
+  "NoDiscovery": false,
```
Comment (adambabik) [inline config/cli/fleet-eth.test.json:3]: Not related: I wonder why this config misses `Rendezvous: true`.

File: cmd/node-canary/main.go [hunk 2]
```diff
@@ -253,6 +253,7 @@ func makeNodeConfig() (*params.NodeConfig, error) {
 		nodeConfig.LogFile = *logFile
 	}
 
+	nodeConfig.NoDiscovery = true
```
Comment (adambabik) [inline cmd/node-canary/main.go:259]: Is it needed? `NewNodeConfigWithDefaults` sets `c.NoDiscovery = true` now and `WithFleet` option is not set.

**Comment by jakubgs** [comment]
Nice, works fine now:
```
% ./build/bin/node-canary -mailserver enode://707e57453acd3e488c44b9d0e17975371e2f8fb67525eae5baca9b9c8e06c86cde7c794a6c2e36203bf9f56cae8b0e50f3b33c4c2b694a7baeea1754464ce4e3
INFO [10-27|15:55:30.148] Status backend initialized 
INFO [10-27|15:55:30.166] LES protocol is disabled                 package=status-go/node
INFO [10-27|15:55:30.166] Status service api is disabled           package=status-go/node
INFO [10-27|15:55:30.167] Starting peer-to-peer node               instance=node-canary/linux-amd64/go1.10.4
INFO [10-27|15:55:30.167] Starting P2P networking 
INFO [10-27|15:55:30.167] started whisper v.6.0 
INFO [10-27|15:55:30.172] Handlers registered                      package=status-go/api.StatusBackend
INFO [10-27|15:55:30.172] Account reselected                       package=status-go/api.StatusBackend
INFO [10-27|15:55:30.227] Difference with ntp servers              offset=918.016µs
ERROR[10-27|15:55:35.173] Error detected while waiting for mailserver peer to be added package=status-go/cmd/node-canary   error="wait for peer: timeout"
```

**Review by adambabik — APPROVED** [review]

**Comment by pedropombeiro** [comment]
I think this PR still needs more work, since I tested with a branch before the latest changes and it didn't timeout with the same server. Some recent change is causing the timeout.

File: cmd/node-canary/main.go (hunk 2 above)
Comment (pedropombeiro) [reply cmd/node-canary/main.go:259]: Yeah, it's only making sure that it's enabled, but we can remove it.

File: config/cli/fleet-eth.test.json (hunk 1 above)
Comment (pedropombeiro) [reply config/cli/fleet-eth.test.json:3]: What are the rendezvous nodes enode addresses we'll want to use here?

**Comment by jakubgs** [comment]
Any news on this? I just stubbed my toe on this while switching https://canary.status.im/ from Slack integration to PagerDuty.

<blockquote><div><strong><a href="https://canary.status.im/">Sign In</a></strong></div></blockquote>

**Pull request merged**